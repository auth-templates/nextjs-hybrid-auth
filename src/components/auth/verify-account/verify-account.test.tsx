import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import VerifyAccount from './verify-account';

describe('VerifyAccount', () => {
    test('if all initial elements are present', () => {
        render(<VerifyAccount/>);
        expect(screen.getByTestId('small-lds-ring')).toBeInTheDocument();
    })

    test('if message box is displayed when an error is received from the server', async () => {
        render(
            <VerifyAccount 
                status={{theme:'error', lines:['Failed verify account. Please try again!']}}
            />
        );

        expect(await screen.getByText(/Failed verify account. Please try again!/)).toBeInTheDocument();
    })

    test('if account verified component is displayed when "Account verified" message is received', async () => {
        render(
            <MemoryRouter>
                <VerifyAccount 
                    status={{theme:'info', lines:['Account verified']}}
                />
            </MemoryRouter>
        );

        expect(await screen.findByText(/Your account has been verified successfully. Please go to/)).toBeInTheDocument();
    })

    test('if account confirmation email expired component is displayed when "Confirmation token is not valid" message is received', async () => {
        render(
            <MemoryRouter>
                <VerifyAccount
                    status={{theme:'error', lines:['Confirmation token is not valid']}}
                />
            </MemoryRouter>
        );

        expect(await screen.findByText(/Your account confirmation email has expired. Please go to/)).toBeInTheDocument();
    })

    it('renders correctly', () => {
        const tree = renderer.create(<VerifyAccount />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})