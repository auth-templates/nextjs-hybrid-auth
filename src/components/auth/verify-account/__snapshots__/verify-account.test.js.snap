// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerifyAccount renders correctly 1`] = `
<div
  className="verifyAccount"
>
  <div
    className="ldsRing smallLoader"
    data-testid="small-lds-ring"
  >
    <div />
    <div />
  </div>
</div>
`;
